src/transformation/transformers/binary/destructuring-assignments.ts

```diff
@@ -0,0 +1,280 @@
+import * as ts from "typescript";
+import * as tstl from "../../../LuaAST";
+import { flatMap } from "../../../utils";
+import { TransformationContext } from "../../context";
+import { UnsupportedKind } from "../../utils/errors";
+import { LuaLibFeature, transformLuaLibFunction } from "../../utils/lualib";
+import { isArrayType, isAssignmentPattern } from "../../utils/typescript";
+import { transformIdentifier } from "../identifier";
+import { transformPropertyName } from "../literal";
+import { transformAssignment, transformAssignmentStatement } from "./assignments";
+
```

Comment:
This binary directory name really confused me for a bit there, I was thinking of binary as bits (since I saw the bitwise operators first), not binary expressions. Would be nice if this can be ranamed to something like binary-expressions. Taking that a bit broader, should we in general indicate whether a folders transformers concern expressions or assignments?

Reply:
I've renamed n-ary transformers directories.

(Assuming you meant expressions or statements) I don't think that would make it better. Transformers are grouped by their semantic meaning, so a lot of them handle both expressions and statements (and sometimes declarations or other node kinds).